auto_gptq/modeling/_utils.py

```diff
@@ -130,6 +130,55 @@ def make_quant(
                 )
             new_layer.device = ori_layer_device
             recurse_setattr(module, name, new_layer.to(ori_layer_device))
+            
+def convert_new_checkpoint_format(
+    model,
+    to_new_format,
+    quantize_config,
+    QuantLinear,
+):
+    use_qigen = QuantLinear.QUANT_TYPE == "qigen"
+    use_marlin = QuantLinear.QUANT_TYPE == "marlin"
+
+    for name, submodule in model.named_modules():
+        if isinstance(submodule, QuantLinear):
+            if to_new_format:
+                if use_qigen:
+                    submodule.zeros.data += 1
+                elif use_marlin:
+                    pass
+                else:
+                    if quantize_config.bits == 2:
+                        submodule.qzeros.data += 0b01010101010101010101010101010101
+                    elif quantize_config.bits == 3:
+                        submodule.qzeros.data[:,range(0,submodule.qzeros.data.shape[1],3)] += 0b00100100100100100100100100100100
+                        submodule.qzeros.data[:,range(1,submodule.qzeros.data.shape[1],3)] += 0b10010010010010010010010010010010
+                        submodule.qzeros.data[:,range(2,submodule.qzeros.data.shape[1],3)] += 0b01001001001001001001001001001001
+                    elif quantize_config.bits == 4:
+                        submodule.qzeros.data += 0b00010001000100010001000100010001
+                    elif quantize_config.bits == 8:
+                        submodule.qzeros.data += 0b00000001000000010000000100000001
```

Comment:
This does not check for overflows, this is an issue. We used to check for overflows in the kernels.

11111 will become 0000.